metadata/environment.go

```diff
@@ -293,12 +293,20 @@ func (m *manager) SetEnvironment(name string, desired *Environment) error {
 		}
 
 		// Move the directory
-		pathOld := string(appendToAbsPath(m.environmentsPath, name))
-		pathNew := string(appendToAbsPath(m.environmentsPath, desired.Name))
-		log.Debugf("Moving directory at path '%s' to '%s'", pathOld, pathNew)
-		err = m.appFS.Rename(pathOld, pathNew)
+		pathOld := appendToAbsPath(m.environmentsPath, name)
+		pathNew := appendToAbsPath(m.environmentsPath, desired.Name)
+		jsonnetPathOld := appendToAbsPath(pathOld, path.Base(name)+".jsonnet")
+		jsonnetPathNew := appendToAbsPath(pathOld, desired.Name+".jsonnet")
+		log.Debugf("Renaming jsonnet file from '%s' to '%s'", path.Base(string(jsonnetPathOld)), path.Base(string(jsonnetPathNew)))
+		err = m.appFS.Rename(string(jsonnetPathOld), string(jsonnetPathNew))
```

Comment:
I think this isn't going to work unless you create the new directory first. Right? So renaming say `dev` to `us-west/dev` will fail. This is tricky so we probably want to add tests, also.

Reply:
No, this will work. afero's `rename` is a bit misleading that it actually does the equivalent of `mv`.

There's also an open issue with afero that isn't able to allow me to add the unit test I want. (Because testFS won't move the file contents despite the actual code working): https://github.com/spf13/afero/issues/141

Reply:
Sorry, what I'm saying is that renaming some environment `bar` -> `us-west/bar` will fail because we're moving the `.jsonnet` file to the `us-west/` directory before creating it, which will fail.

I've tested and confirmed:

```
➜  simple git:(pr-58) ✗ ../ks env set bar --name us-west/dev
INFO  Setting environment name from 'bar' to 'us-west/dev'
ERROR rename /Users/alex/src/go/src/github.com/ksonnet/ksonnet/simple/environments/bar/bar.jsonnet /Users/alex/src/go/src/github.com/ksonnet/ksonnet/simple/environments/bar/us-west/dev.jsonnet: no such file or directory
```